appservice/src/deleteSite.ts

```diff
@@ -22,15 +22,15 @@ export async function deleteSite(client: SiteClient): Promise<void> {
         plan = await client.getAppServicePlan();
     }
 
-    if (!client.isSlot && plan.numberOfSites < 2) {
+    if (!client.isSlot && plan && plan.numberOfSites < 2) {
```

Comment:
Believe this might be the reason for the numberOfSites error I was seeing.  Once again, Eric's point about non-null assertion stands.  My guess is that client.getAppServicePlan was returning 'null' because of some timing issue between creating the asp and then immediately trying to get it.  Which would explain why sometimes it would work and other times it would not.